agent/cache-types/catalog_services_test.go

```diff
@@ -41,6 +42,7 @@ func TestCatalogServices(t *testing.T) {
 		Datacenter:  "dc1",
 		ServiceName: "web",
 		ServiceTag:  "canary",
```

Comment:
Can we remove this line please? Internally we never use this field anymore. The change to this test doesn't actually assert anything about the cache but it is meant to just keep consistency internally that we expect the `ServiceTags` to be populated not deprecated `ServiceTag` now.

I think this is enough to merge with since the change is simple but a nice to have for the future would be to extend this test to actually request the same thing twice with different tag sets and assert they get different responses. There are example of this for tokens in the ConnectLeaf cache type IIRC.